File: robot/Comm/Messages/Remote/FromRobot/LogEndMessage.cpp
#include <string.h>

#include "LogEndMessage.h"

LogEndMessage::LogEndMessage()
  : RemoteMessageFromRobot(RemoteMessageFromRobot::logEnd, 0, NULL) {
  createData();
}

LogEndMessage::~LogEndMessage() {
}

void LogEndMessage::createData() {
  data = new char[1];
  if (data == NULL) {
    size = 0;
    return;
  }

  data[0] = 0;
  size = 1;
}

File: tools/SharedJavaLibrary/src/network/message/fromrobot/LogEndMessage.java
package network.message.fromrobot;

import network.message.RemoteMessageFromRobot;

public class LogEndMessage extends RemoteMessageFromRobot {
  
  public LogEndMessage() {
    super(RemoteMessageFromRobot.MessageType.LOG_BEGIN, null);
  }
  
  public static LogEndMessage create(byte [] data) {
    try {
      return new LogEndMessage();
    }
    catch (Exception e) {
      return null;
    }
  }

}
